Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/GasArtifactSystem.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Content.Server.Atmos;
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Components;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Events;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Systems;
+
+public class GasArtifactSystem : EntitySystem
+{
+    [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GasArtifactComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<GasArtifactComponent, ArtifactActivatedEvent>(OnActivate);
+    }
+
+    private void OnInit(EntityUid uid, GasArtifactComponent component, ComponentInit args)
+    {
+        if (component.SpawnGas == null && component.PossibleGases.Length != 0)
+        {
+            var gas = _random.Pick(component.PossibleGases);
+            component.SpawnGas = gas;
+        }
+
+        if (component.SpawnTemperature == null)
+        {
+            var temp = _random.NextFloat(component.MinRandomTemperature, component.MaxRandomTemperature);
+            component.SpawnTemperature = temp;
+        }
+    }
```

Comment:
Forgot to mention it on the last PR but these should be MapInit so it's randomised every time IMO.

Reply:
I moved all of them, except this one
```
    private void OnInit(EntityUid uid, ArtifactComponent component, ComponentInit args)
    {
        if (component.RandomTrigger)
        {
            AddRandomTrigger(uid, component);
        }
    }

    public void AddRandomTrigger(EntityUid uid, ArtifactComponent? component = null)
    {
        if (!Resolve(uid, ref component))
            return;

        var triggerName = _random.Pick(component.PossibleTriggers);
        var trigger = (Component) _componentFactory.GetComponent(triggerName);
        trigger.Owner = uid;

        EntityManager.AddComponent(uid, trigger);
    }
```

If I try to use `MapInit` event here - I'll get an exception. The reason is that some triggers use `MapInit` too. I can work around of it with deferred spawning using update loop, but...

Reply:
you could fix this by having only this one work on mapinit, then maybe have the other ones use init (since theyre only added after mapinit anyway?)

Reply:
> you could fix this by having only this one work on mapinit, then maybe have the other ones use init (since theyre only added after mapinit anyway?)

Yes, but sloth said to convert all random behavior to MapInit. If I'll use just init, reloading maps will generate random values each time.

The only sane solution is use timer, to skip one frame. But they are restricted.

Reply:
If the triggers are only on activate then you could defer having one added until first interaction maybe?

Most of the time artifacts are probably going to be made via spawners and not saved directly to a map I would think so it's probably not an issue in most cases.

Reply:
In the end, I added separate event `RandomizeTriggerEvent`. It's raised when artifact adds random trigger. Trigger components randomize their data without map init.

Reply:
sounds fine